nomos-blend/network/src/conn_maintenance.rs

```diff
@@ -79,7 +79,7 @@ where
         let Poll::Ready(Some(new_expected_message_count_range)) =
             self.connection_window_clock.poll_next_unpin(cx)
         else {
-            cx.waker().wake_by_ref();
+            // Waker will be called by the `poll_next_unpin` when a new item is ready.
```

Comment:
who's going to call `poll_next_unpin` if the above call returns `Poll:Ready(None))`

Reply:
When `poll_next_unpin` is called, a context with `waker` is passed to the stream handler, stream calls waker when item is ready.

Reply:
Oh if it's `Poll::Ready(None)` then it executes this poll method further and it's up tho this whole `poll` method to decide what happens next.

Reply:
So looked at the docs - when stream returns `Poll::Ready(None)` - it doesn't call waker anymore, stream is closed. Not sure what should happen then. Probably Poll::Pending, effectively never executing poll method again?

Reply:
Yes, I answered something similar to a similar point in a related PR. I think it's fine to return `Ready(None)` if any of the substreams returns `Ready(None)` indeed, and let the caller decide whether to panic or do something else.

Reply:
oh right, I missed the part where `ConnectionWindowClock` is a stream, all good to return `Ready(None)` in this case, as we shouldn't schedule it anymore